addons/mail/tests/discuss/test_discuss_channel_as_guest.py

```diff
@@ -112,3 +113,61 @@ def test_sidebar_in_public_page(self):
         channel_1._add_members(guests=guest)
         channel_2._add_members(guests=guest)
         self.start_tour(f"/discuss/channel/{channel_1.id}", "sidebar_in_public_page_tour", cookies={guest._cookie_name: guest._format_auth_cookie()})
+
+    def test_updating_guest_name_linked_to_multiple_channels(self):
+        guest = self.env['mail.guest'].create({'name': 'Guest'})
+        channel_1 = self.env["discuss.channel"]._create_channel(name="Channel 1", group_id=None)
+        channel_2 = self.env["discuss.channel"]._create_channel(name="Channel 2", group_id=None)
+        channel_1._add_members(guests=guest)
+        channel_2._add_members(guests=guest)
+        guest_write_date = fields.Datetime.to_string(guest.write_date)
+        self._reset_bus()
+        with self.assertBus(
```

Comment:
There can be a race condition with the write_date if it's read before doing the update.

The test should use the callback param of assertBus instead so the value is read after the action.